packages/plugin/src/interpreter.ts

```diff
@@ -105,7 +114,11 @@ export class Interpreter {
         }
 
         let optionsTypeName: string;
-        if (FieldDescriptorProto.is(descriptor) && DescriptorProto.is(this.registry.parentOf(descriptor))) {
+        if (knownOptionTypeName) {
+            optionsTypeName = `google.protobuf.${knownOptionTypeName}`;
+        } else if (OneofDescriptorProto.is(descriptor) && DescriptorProto.is(this.registry.parentOf(descriptor))) {
+            optionsTypeName = 'google.protobuf.OneofOptions';
+        } else if (FieldDescriptorProto.is(descriptor) && DescriptorProto.is(this.registry.parentOf(descriptor))) {
```

Comment:
Unfortunately the way `MessageType#is` behaves (assignable and no excess properties) means that the order of these if statements is extremely important. I spent an hour trying to figure out why the oneof options weren't being populated when I put the `OneofDescriptorProto.is(descriptor)` check after the `FieldDescriptorProto.is(descriptor)` check. Well, it's because `FieldDescriptorProto.is(oneofDescriptor) && DescriptorProto.is(this.registry.parentOf(oneofDescriptor))` returns true.

One way to work around this is what I've done in this PR is to have the caller just tell us which option type name goes with the descriptor. But I've also realized that we can just use the `MESSAGE_TYPE` symbol on the `descriptor` to know which descriptor `MessageType` it is and therefore which option type name should be used so I'm going to switch to using that.